src/backend/base/langflow/components/tools/mcp_component.py

```diff
@@ -426,8 +426,12 @@ async def build_output(self) -> Message:
 
                 output = await exec_tool.coroutine(**unflattened_kwargs)
 
-                return Message(text=output.content[len(output.content) - 1].text)
-            return Message(text="You must select a tool", error=True)
+                tool_content = []
+                for item in output.content:
+                    item_dict = item.model_dump()
+                    tool_content.append(item_dict)
+                return DataFrame(data=tool_content)
+            return DataFrame(data=[{"error": "You must select a tool"}])
```

Comment:
```suggestion
            return DataFrame()
```

This would be the ideal way to go. if not 
raise ValueError("Tool not defined, Select a Tool before execution")